packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
They would, but they're going to have to switch on platform in their app code anyway in order to do anything useful with the result since trying to get the iOS info on Android will throw and vice versa.  So if they're already a device-specific `if` block, the "cast" will simply look like:

```
if (defaultTargetPlatform == TargetPlatform.android) {
  AndroidDeviceInfo info = DeviceInfoPlugin.info;
  // Extract Android info
} else if (defaultTargetPlatform == TargetPlatform.iOS) {
  IosDeviceInfo info = DeviceInfoPlugin.info;
  // Extract iOS info
}
```